spring-cloud-azure-starters/spring-cloud-starter-azure-appconfiguration-config/README.md

```diff
@@ -37,77 +40,126 @@ spring.cloud.azure.appconfiguration.fail-fast | Whether throw RuntimeException o
 By default, all the keys in a configuration store will be watched. To prevent configuration changes are picked up in the middle of an update of multiple keys, you are recommended to use the watched-key property to watch a specific key that signals the completion of your update so all configuration changes can be refreshed together.
-```
+
+```properties
 spring.cloud.azure.appconfiguration.stores[0].watched-key=[my-watched-key]
 ```
 
 ### Failfast
+
 Failfast feature decides whether throw RuntimeException or not when exception happens. By default, failfast is enabled, it can be disabled with below configuration:
-```
+
+```properties
 spring.cloud.azure.appconfiguration.fail-fast=false
 ```
 
-### Use Azure Managed Identity to load the connection string
-[Managed service identity](https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/overview) allows application to access [Azure Active Directory](https://azure.microsoft.com/en-us/services/active-directory/) protected resource on [Azure](https://azure.microsoft.com/en-us/).
+### Use Azure Managed Identity/Service Principle to load the connection string
+
+[Managed service identity](https://docs.microsoft.com/azure/active-directory/managed-identities-azure-resources/overview) allows application to access [Azure Active Directory](https://azure.microsoft.com/services/active-directory/) protected resource on [Azure](https://azure.microsoft.com).
 
-In this library, managed service identity is used to retrieve the connection string of the configuration store, the connection string is not required if running the Spring Boot application on Azure with managed service identity enabled.
+In this library, [Azure Identity SDK][azure_identity_sdk] is used to access Azure App Configuration and optionally Azure Key Vault, for secrets. The connection string is not required and will be ignored if Managed Identity is being used.
 
 Follow below steps to enable managed service identity feature:
 
-1. [Enable managed identities service](https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/overview#how-can-i-use-managed-identities-for-azure-resources) for virtual machine or App Service, on which the application will be deployed
+1. [Enable managed identities service](https://docs.microsoft.com/azure/active-directory/managed-identities-azure-resources/overview#how-can-i-use-managed-identities-for-azure-resources) for virtual machine or App Service, on which the application will be deployed
+
+1. Configure the [Azure RBAC](https://docs.microsoft.com/azure/role-based-access-control/role-assignments-portal) to allow application running on VM or App Service to access the configuration store. When prompted for the Role select App Configuration Data Reader, App Configuration Data Owner is not required but can be used if needed.
```

Comment:
Configure the [Azure RBAC](https://docs.microsoft.com/azure/role-based-access-control/role-assignments-portal) of your App Configuration store to grant access to the Azure service where your application is running. Select the *App Configuration Data Reader* role. The *App Configuration Data Owner* role is not required but can be used if needed.